tests/MongoDB.Driver.Examples/CausalConsistencyExamples.cs

```diff
@@ -0,0 +1,130 @@
+                    {"start", currenDate}
+                });
+            }
+            //End Causal Consistency Example 1
+
+            var result = testDatabase.GetCollection<BsonDocument>(itemsCollectionName).Find("{}").FirstOrDefault();
+            RemoveIds(new[] { result });
+            result["start"].Should().NotBeNull();
+            result.Remove("start");
+            result.Should().Be("{ \"sku\" : \"nuts-111\", \"name\" : \"Pecans\" }");           
+        }
+
+        [Fact]
+        public void Causal_Consistency_Example_2()
+        {
+            RequireServer.Check().SupportsCausalConsistency();
+
+            var client = GetClient();
+
+            string itemsCollectionName = "items";
+            var testDatabase = client.GetDatabase("test");
+            testDatabase.DropCollection(itemsCollectionName);
+
+            using (var session1 = client.StartSession(new ClientSessionOptions { CausalConsistency = true }))
+            {
+                var currenDate = DateTime.UtcNow.Date;
+                var items = client.GetDatabase("test", new MongoDatabaseSettings
+                    {
+                        ReadConcern = ReadConcern.Majority,
+                        WriteConcern = new WriteConcern(
+                            WriteConcern.WMode.Majority,
+                            TimeSpan.FromMilliseconds(1000))
+                    })
+                    .GetCollection<BsonDocument>("items");
+
+                items.InsertOne(session1, new BsonDocument
+                {
+                    {"sku", "nuts-111"},
+                    {"name", "Pecans"},
+                    {"start", currenDate}
+                });
+
+                // Start Causal Consistency Example 2
+                using (var session2 = client.StartSession(new ClientSessionOptions { CausalConsistency = true }))
+                {
+                    session2.AdvanceClusterTime(session1.ClusterTime);
+                    session2.AdvanceOperationTime(session1.OperationTime);
+
+                    var items2 = client.GetDatabase("test", new MongoDatabaseSettings
+                        {
+                            ReadPreference = ReadPreference.Secondary,
+                            ReadConcern = ReadConcern.Majority,
+                            WriteConcern = new WriteConcern(WriteConcern.WMode.Majority, TimeSpan.FromMilliseconds(1000))
+                        })
+                        .GetCollection<BsonDocument>("items");
+
+                    var filter = Builders<BsonDocument>.Filter.Eq("end", BsonNull.Value);
+                    foreach (var item in items2.Find(session2, filter).ToEnumerable())
+                    {
+                        var print = item;
+                    }
+                }
+                // End Causal Consistency Example 2
+            }
+
+            var result = testDatabase.GetCollection<BsonDocument>(itemsCollectionName).Find("{}").FirstOrDefault();
```

Comment:
I think you can remove this from Example 2. This is testing the code from Example 1, not from Example 2.

Reply:
So in this case, we will not have any assert checkings for the example 2?
At least if we still use `using` for session2